src/UrlGeneratorInterface.php

```diff
@@ -45,13 +45,11 @@ public function generateAbsolute(
         string $host = null
     ): string;
 
+    public function generateFromCurrent(array $replacedParams, string $fallbackRouteName = null): string;
+
     public function getUriPrefix(): string;
 
     public function setUriPrefix(string $name): void;
 
-    public function getLocales(): array;
-
-    public function setLocales(array $locales): void;
-
-    public function setLocaleParameterName(string $localeParameterName): void;
+    public function setDefault(string $name, $value): void;
```

Comment:
May be allow setmultiple parameters as `setDefaults(array $defaults)` ?
This is will be consistent: in other places of this packages we set arrays of parameters/defaults.

Reply:
No, there are many places where we can set different defaults. The one method overrides previous defaults.